R/string_boundary_linter.R

```diff
@@ -21,6 +21,35 @@
 #'   are ignored. Some authors may prefer the conciseness offered by `grepl()` whereby
 #'   `NA` input maps to `FALSE` output, which doesn't have a direct equivalent
 #'   with `startsWith()` or `endsWith()`.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = 'grepl("^a", x)',
+#'   linters = string_boundary_linter()
+#' )
+#'
+#' lint(
+#'   text = 'grepl("z$", x)',
+#'   linters = string_boundary_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = 'startsWith(x, "a")',
+#'   linters = string_boundary_linter()
+#' )
+#'
+#' lint(
+#'   text = 'endsWith(x, "z")',
+#'   linters = string_boundary_linter()
+#' )
+#'
+#' lint(
+#'   text = 'grepl("z$", x)',
```

Comment:
this is not very illustrative of _why_ this argument exists. would be more instructive to mention about missing behavior